main.cpp

```diff
@@ -0,0 +1,42 @@
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "Unit.h"
+
+void const printUnits(Unit UnitA, Unit UnitB) {
+    std::cout << UnitA.getName() << " Hp:" << UnitA.getHealth() << " Damage: " << UnitA.getDamage() << std::endl;
+    std::cout << UnitB.getName() << " Hp:" << UnitB.getHealth() << " Damage: " << UnitB.getDamage() << std::endl;
+}
+
+void fight(Unit& UnitA, Unit& UnitB){
+    printUnits(UnitA,UnitB);
+    while(UnitA.getHealth() > 0 && UnitB.getHealth() > 0){
+        std::cout << UnitA.getName() << " -> " << UnitB.getName() << std::endl; 
+        UnitB.getAttacked(UnitA.getDamage());
```

Comment:
A jelen szignaturaval itt akar beirhatsz veletlenul 2*UnitA.getDamage()-et is. Az ilyen hibakat volna jo elkerulni.(tobbek kozott)